src/gsy_e/models/strategy/energy_parameters/energy_params_eb.py

```diff
@@ -21,37 +21,190 @@
+            slots = self.day_forward_slots(market_slot)
+            for slot in slots:
+                if slot not in self._posted_energy_kWh:
+                    self._posted_energy_kWh[slot] = 0.
+                self._posted_energy_kWh[slot] += posted_energy_kWh
+
+    def decrement_posted_energy(
+            self, market_slot: pendulum.DateTime, posted_energy_kWh: float,
+            market_type: AvailableMarketTypes):
+        """
+        Decrease the posted energy of the strategy. Needs to handle only intraday and day ahead
+        since these are the only 2 market types that can operate at the same market slots
+        concurrently.
+        """
+        if market_type not in [AvailableMarketTypes.INTRADAY, AvailableMarketTypes.DAY_FORWARD]:
+            return
+
+        assert market_slot in self._posted_energy_kWh, \
+            "Failed to decrement posted energy, the posted energy should be updated on every " \
+            "bid or offer before decrementing it."
+
+        if market_type == AvailableMarketTypes.INTRADAY:
+            self._posted_energy_kWh[market_slot] -= posted_energy_kWh
+        if market_type == AvailableMarketTypes.DAY_FORWARD:
+            slots = self.day_forward_slots(market_slot)
+            for slot in slots:
+                self._posted_energy_kWh[slot] -= posted_energy_kWh
+
+    @abstractmethod
+    def serialize(self):
+        """Return dict with the current energy parameter values."""
+
+    @abstractmethod
+    def get_available_energy_kWh(
+            self, market_slot: pendulum.DateTime, market_type: AvailableMarketTypes):
+        """Get the available offer energy of the PV."""
+
+    @abstractmethod
+    def event_activate_energy(self, area):
+        """Initialize values that are required to compute the energy values of the asset."""
+
+    @abstractmethod
+    def event_traded_energy(
+            self, energy_kWh: float, market_slot: pendulum.DateTime,
+            product_type: AvailableMarketTypes):
+        """
+        When a trade happens, we want to split the energy through the entire year following a
+        standard solar profile.
+
+        Args:
+            energy_kWh: the traded energy in kWh.
+            market_slot: the slot targeted by the trade. For forward exchanges, this represents
+                the entire period of time over which the trade profile will be generated.
+            product_type: One of the available market types.
+        """
+
+    @property
+    @abstractmethod
+    def state(self) -> "StateInterface":
+        """Retrieve the internal state of the asset."""
+
+    @staticmethod
+    def day_forward_slots(market_slot: pendulum.DateTime):
+        """Get the market slots for the day forward market."""
+        return [
+            market_slot,
+            market_slot + pendulum.duration(minutes=15),
+            market_slot + pendulum.duration(minutes=30),
+            market_slot + pendulum.duration(minutes=45)]
```

Comment:
So the DayForward is really only a HourForward any more right?

Reply:
Yes, in order to be similar to the day-ahead EPEX spot market. Essentially the market slots are hourly, but expand over the next day. Therefore there are 24 slots from 00:00 till 23:45 of the next day.

Reply:
I do not get it. I see here a list of 4 market slots, not 24.

Reply:
Probably the comment is misleading here. What you see here is the 15-min times that one DayForward market slot includes. Since the duration of a DayForward slot is 1 hour, and the market slot is e.g. today 12:00, then there will be 4 entries in the profile affected by this market: 12:00, 12:15, 12:30 and 12:45. I struggle to find a better explanation though for this in order to reflect it in a comment.